src/test/java/jssc/junit/rules/DisplayMethodNameRule.java

```diff
@@ -0,0 +1,22 @@
+package jssc.junit.rules;
+
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.apache.logging.log4j.MarkerManager;
+import org.junit.Rule;
+import org.junit.rules.TestWatcher;
+import org.junit.runner.Description;
+
+/**
+ * Adds the method name to the JUnit logs, useful for debugging
+ */
+public class DisplayMethodNameRule {
+    @Rule
+    public TestWatcher methodWatcher = new TestWatcher() {
+        @Override
+        protected void starting(Description description) {
+            Logger log = LogManager.getLogger(description.getTestClass());
+            log.debug(MarkerManager.getMarker("MethodName"), description.getMethodName());
```

Comment:
Use Log4j's marker to help logger distinguish between regular logs and this special type, for which we want to apply special, maven-like formatting.

Reply:
Thank you, this works very well!